vsphere/tests/test_event.py

```diff
@@ -291,3 +291,61 @@ def test_include_events_ok_new_event_type(aggregator, realtime_instance):
     assert len(aggregator.events) == 1
     assert "The Alarm was acknowledged" in aggregator.events[0]['msg_text']
     assert aggregator.events[0]['msg_title'] == "AlarmAcknowledgedEvent"
+
+
+@pytest.mark.usefixtures('mock_type', 'mock_threadpool', 'mock_api', 'mock_rest_api')
+def test_include_events_ok_new_resource(aggregator, realtime_instance):
+    realtime_instance['include_events'] = [{"event": "AlarmAcknowledgedEvent", "excluded_messages": ["Remove Alarm"]}]
+    realtime_instance['event_resource_filters'] = ['storage_pod']
+    check = VSphereCheck(
+        'vsphere',
+        {},
+        [realtime_instance],
+    )
+    check.initiate_api_connection()
+
+    event1 = vim.event.AlarmAcknowledgedEvent()
+    event1.createdTime = dt.datetime.now()
+    event1.entity = vim.event.ManagedEntityEventArgument()
+    event1.entity.entity = vim.StoragePod(moId="pod1")
+    event1.entity.name = "pod1"
+    event1.alarm = vim.event.AlarmEventArgument()
+    event1.alarm.name = "alarm1"
+    event1.datacenter = vim.event.DatacenterEventArgument()
+    event1.datacenter.name = "dc1"
+    event1.fullFormattedMessage = "The Alarm was acknowledged"
+    aggregator.reset()
+    check.api.mock_events = [event1]
+    check.check(None)
+    assert len(aggregator.events) == 1
+    assert "The Alarm was acknowledged" in aggregator.events[0]['msg_text']
+    assert aggregator.events[0]['msg_title'] == "AlarmAcknowledgedEvent"
+
+
+@pytest.mark.usefixtures('mock_type', 'mock_threadpool', 'mock_api', 'mock_rest_api')
+def test_include_events_excluded_message_new_resource(aggregator, realtime_instance):
+    realtime_instance['include_events'] = [
+        {"event": "AlarmAcknowledgedEvent", "excluded_messages": ["The Alarm was acknowledged"]}
+    ]
+    realtime_instance['event_resource_filters'] = ['storage_pod']
+    check = VSphereCheck(
+        'vsphere',
+        {},
+        [realtime_instance],
+    )
+    check.initiate_api_connection()
+
+    event1 = vim.event.AlarmAcknowledgedEvent()
+    event1.createdTime = dt.datetime.now()
+    event1.entity = vim.event.ManagedEntityEventArgument()
+    event1.entity.entity = vim.StoragePod(moId="pod1")
+    event1.entity.name = "pod1"
+    event1.alarm = vim.event.AlarmEventArgument()
+    event1.alarm.name = "alarm1"
+    event1.datacenter = vim.event.DatacenterEventArgument()
+    event1.datacenter.name = "dc1"
+    event1.fullFormattedMessage = "The Alarm was acknowledged"
+    aggregator.reset()
+    check.api.mock_events = [event1]
+    check.check(None)
+    assert len(aggregator.events) == 0
```

Comment:
Can we have a test for when `event_resource_filters` is not set?

Reply:
[By default, the integration will only collect these events for VMs and Hosts.](https://github.com/DataDog/integrations-core/blob/master/vsphere/datadog_checks/vsphere/data/conf.yaml.example#L253)

However, I added a test where `event_resource_filters` is specified as empty, to make sure that none of the 'PER_RESOURCE_EVENTS' were collected.

Reply:
that's perfect!